libs/custom-paper-handler/src/driver/driver.ts

```diff
@@ -369,27 +375,65 @@ export class PaperHandlerDriver implements PaperHandlerDriverInterface {
     const transferOutResult = await this.transferOutGeneric(coder, value);
     assert(transferOutResult.status === 'ok'); // TODO: Handling
 
+    return this.transferInAcknowledgement();
+  }
+
+  async transferInAcknowledgement(): Promise<boolean> {
     const transferInResult = await this.transferInGeneric();
     assert(transferInResult.status === 'ok'); // TODO: Handling
     this.genericLock.release();
     const { data } = transferInResult;
     assert(data);
-    const result = AcknowledgementResponse.decode(Buffer.from(data.buffer));
+
+    // If data buffered to the `generic transfer-out buffer` contains the command code
+    // for `Real-time status transmission` (0x10 0x04 n), that command will be executed.
+    // It is executed regardless of whether the buffered data is received as part of a
+    // different command and ignores convention that `Real-time status transmission` is
+    // expected to use the real-time channels.
+    //
+    // eg. `bufferChunk()` uses `Select image print mode` to buffer PDF data to the
+    // generic out buffer. If the PDF data contains the byte sequence `0x10 0x04 0x12`,
+    // the device will execute `Real-time status transmission` and return a 6 byte response
+    // on the generic transfer-in buffer.
+    //
+    // If the PDF data contains byte sequence `0x10 0x04 0x03`, where 0x03 is an invalid
+    // argument for `Real-time status transmission`, the printer will return 1 byte 0x12
+    // on the generic transfer-in buffer. 0x12 is an undocumented response code but
+    // presumably means "Invalid argument" or similar.
+
+    const TransferInData = oneOf(
+      RealTimeStatusTransmission,
+      AcknowledgementResponse
+    );
+    const result = TransferInData.decode(Buffer.from(data.buffer));
     if (result.isErr()) {
       debug(`Error decoding transferInGeneric response: ${result.err()}`);
       return false;
     }
+
     const code = result.ok();
-    switch (code) {
-      case ReturnCodes.POSITIVE_ACKNOWLEDGEMENT:
-        debug('positive acknowledgement');
-        return true;
-      case ReturnCodes.NEGATIVE_ACKNOWLEDGEMENT:
-        debug('negative acknowledgement');
-        return false;
-      default:
-        throw new Error(`uninterpretable acknowledgement: ${code}`);
+    if (typeof code === 'number') {
+      switch (code) {
+        case ReturnCodes.POSITIVE_ACKNOWLEDGEMENT:
+          debug('positive acknowledgement');
+          return true;
+        case ReturnCodes.NEGATIVE_ACKNOWLEDGEMENT:
+          debug('negative acknowledgement');
+          return false;
+        case ReturnCodes.INVALID_STATUS_TRANSMISSION_ARGUMENT:
```

Comment:
Is it possible that a blob might contain a _valid_ status transmission request? Would we need to handle those responses as well here?

Reply:
This is already handled - @kshen0 walked me through it in person

That said, we're now taking a step back and leaning toward the likely safer alternative of just always reading/clearing the in buffer after the print command, instead of catching unexpected responses from a previous print command while reading the response for a subsequent command

Reply:
This makes to me sense provided we aren't worried about other endpoints accidentally triggering the same command.

Reply:
For the record it is possible to send a well-formed status transmission request `0x10 0x04 0x14` and we do handle it appropriately.

We looked into reading the buffer after every print command, but with the WebUSB API once a `transferInGeneric` is initiated it can't be canceled. So in the vast majority of cases we would block on the `transferInGeneric` command.

We can avoid blocking the caller with standard use of `Promise.race` as we do with `clearGenericInBuffer`. But that still initiates a `transferInGeneric` request, so the next response to a valid command over that buffer will be read and discarded by the `transferInGeneric` request that "timed out"`*`

For simplicity I believe it's best to `transferInGeneric` only when we are guaranteed a response on the transfer-in buffer, and handle unexpected extra data as it comes up.

`*` Example in more detail:

* `printChunk` is called with unproblematic data
* Initiate a `transferInGeneric` (call this `transfer in A`) that unblocks the caller after a timeout but does not (cannot) cancel the transfer in request
* After timeout, state machine continues to issue next generic command `setScanDirection('backward')`
  * This and every other generic command involves 2 steps: issuing the command over the generic transfer out buffer and reading the response over the generic transfer in buffer (call this `transfer in B`)
* After execution of `setScanDirection`, device responds on transfer in buffer with `0x06`
* `transfer in A` request resolves by reading `0x06` from the transfer in buffer. The buffer is now empty
* `transfer in B` is now blocked until the subsequent command responds over the transfer in buffer

Reply:
> This makes to me sense provided we aren't worried about other endpoints accidentally triggering the same command.

The only command that sends arbitrary data is `bufferChunk` and its caller `printChunk`.